theano/sandbox/cuda/basic_ops.py

```diff
@@ -3044,6 +3044,24 @@ def make_node(self, *axis_and_tensors):
         as_tensor_variable_args = [as_cuda_ndarray_variable(x)
                                    for x in tensors]
 
+        # Get joining axis as int
+        axis_int = 0
```

Comment:
This is not a complete fix. If the axis is a constant, it is the best fix I can think of. It will help merge operation in the graph.

But if the axis is symbolic, it will still crash. So you should add test for that case (small modif to the test T_Join_and_Split.test_join_matrixV in tensor/tests/test_basic.py, it is reused by the GPU in cuda/tests/test_basic_ops.py:T_Join_and_Split.test_join_matrixV). After the test is added, you need to modify the perform and the c_code of this op to handle that correctly.